File: src/backend/src/main/java/br/com/grupopipa/gestaointegrada/atendimento/lancamento/dto/LancamentoFinanceiroDTO.java
package br.com.grupopipa.gestaointegrada.atendimento.lancamento.dto;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

import br.com.grupopipa.gestaointegrada.atendimento.convenio.entity.ConvenioTipoCobrancaEnum;
import br.com.grupopipa.gestaointegrada.atendimento.lancamento.entity.LancamentoFinanceiroSituacaoEnum;
import br.com.grupopipa.gestaointegrada.atendimento.lancamento.entity.LancamentoFinanceiroStatusFinanceiroEnum;
import br.com.grupopipa.gestaointegrada.core.dto.DTO;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Builder
public class LancamentoFinanceiroDTO implements DTO {

    private UUID id;

    private UUID atendimentoId;
    private Long atendimentoNumero;
    private LocalDate dataAtendimento;

    private UUID pacienteId;
    private String pacienteNome;

    private UUID convenioId;
    private String convenioNome;
    private ConvenioTipoCobrancaEnum convenioTipoCobranca;

    private BigDecimal valorTotal;
    private LancamentoFinanceiroSituacaoEnum situacao;
    private LancamentoFinanceiroStatusFinanceiroEnum statusFinanceiro;

    private List<LancamentoFinanceiroProcedimentoDTO> procedimentos;

    private String observacoes;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private String createdBy;
    private String updatedBy;
}

File: src/frontend/src/app/components/atendimento/lancamento/model/lancamento-financeiro-dto.ts
import { Exclude, Expose } from 'class-transformer';
import { LancamentoFinanceiroSituacao } from './lancamento-financeiro-situacao.enum';
import { LancamentoFinanceiroStatusFinanceiro } from './lancamento-financeiro-status-financeiro.enum';
import { LancamentoFinanceiroProcedimentoDTO } from './lancamento-financeiro-procedimento-dto';
import { ConvenioTipoCobranca } from '../../convenio/model/convenio-tipo-cobranca.enum';

@Exclude()
export class LancamentoFinanceiroDTO {
  @Expose() id?: string;

  @Expose() atendimentoId?: string;
  @Expose() atendimentoNumero?: number;
  @Expose() dataAtendimento?: string;

  @Expose() pacienteId?: string;
  @Expose() pacienteNome?: string;

  @Expose() convenioId?: string;
  @Expose() convenioNome?: string;
  @Expose() convenioTipoCobranca?: ConvenioTipoCobranca;

  @Expose() valorTotal?: number;
  @Expose() situacao?: LancamentoFinanceiroSituacao;
  @Expose() statusFinanceiro?: LancamentoFinanceiroStatusFinanceiro;

  @Expose() procedimentos?: LancamentoFinanceiroProcedimentoDTO[];

  @Expose() observacoes?: string;

  @Expose() setorId?: string;
  @Expose() setorNome?: string;
  @Expose() unidadeNegocioId?: string;
  @Expose() unidadeNegocioNome?: string;
  @Expose() tituloId?: string;

  @Expose() createdAt?: Date;
  @Expose() updatedAt?: Date;
  @Expose() createdBy?: string;
  @Expose() updatedBy?: string;
}
